packages/playwright-core/src/server/injected/clock.ts

```diff
@@ -262,67 +312,26 @@ export class ClockController {
       else if (timer.type === TimerType.IdleCallback)
         args = [{ didTimeout: false, timeRemaining: () => 0 }];
 
-      timer.func.apply(null, args);
-      await new Promise<void>(f => this._embedder.postTask(f));
+      let error: Error | undefined;
+      try {
+        timer.func.apply(null, args);
+      } catch (e) {
+        error = e;
+      }
+      await new Promise<void>(f => this._embedder.setTimeout(f));
+      return { timerFound: true, error };
     } finally {
       this._duringTick = false;
```

Comment:
Perhaps we should reset `_duringTick` before calling into `embedder.setTimeout()` in this method?